controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1033,7 +1049,7 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
 
                     // idempotent check
                     if (!lastRecord.isPartial()) {
-                        if (lastRecord.getSegments().stream().noneMatch(sealedSegments::contains) &&
+                        if (lastRecord.getSegments().stream().noneMatch(x -> sealedSegments.keySet().contains(x)) &&
```

Comment:
you can do `sealedSegments::containsKey`

Reply:
done